<?xml version="1.0" ?>
<!--

Example usage is with medfield_ft8_2020-11-06-16-12-43.bag
https://www.dropbox.com/sh/rvrlsqrcxyk8glo/AAAUHzQ8bnJqWPIU-4g7t1bZa/phoenix_stack/2020-11-06%20Medfield?dl=1

Example usage:

   $ roslaunch launch/uncompress_image_bag.launch  inbag:=$(pwd)/MIT-Jackal-Phoenix-Medfield/medfield_ft8_2020-11-06-16-12-43.bag outbag:=$(pwd)/MIT-Jackal-Phoenix-Medfield/medfield_ft8_2020-11-06-16-12-43-uncompressed.bag

-->
<launch>
  <arg name="inbag" />
  <arg name="outbag" />

  <node name="rosbag_play" pkg="rosbag"  type="play" args="--clock --delay 2 $(arg inbag)" required="true" />
  <node name="uncompress_color" pkg="image_transport" type="republish" args=" compressed">
    <remap from="in" to="acl_jackal/forward/color/image_raw" />
    <remap from="out" to="acl_jackal/forward/color/image_uncompressed" />
  </node>
  <node name="uncompress_infra1" pkg="image_transport" type="republish" args=" compressed">
    <remap from="in" to="acl_jackal/forward/infra1/image_rect_raw" />
    <remap from="out" to="acl_jackal/forward/infra1/image_uncompressed" />
  </node>
  <node name="rosbag_record" pkg="rosbag" type="record" args=" /tf /tf_static acl_jackal/forward/color/image_uncompressed acl_jackal/forward/infra1/image_uncompressed acl_jackal/microstrain/data acl_jackal/forward/imu -O $(arg outbag)" /> 
</launch>
